File: app/src/main/java/org/xdty/callerinfo/di/modules/MainBottomModule.java
package org.xdty.callerinfo.di.modules;

import org.xdty.callerinfo.contract.MainBottomContact;
import org.xdty.callerinfo.presenter.MainBottomPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class MainBottomModule {

    private MainBottomContact.View mView;

    public MainBottomModule(MainBottomContact.View view) {
        mView = view;
    }

    @Provides
    MainBottomContact.View provideView() {
        return mView;
    }

    @Provides
    MainBottomContact.Presenter providePresenter() {
        return new MainBottomPresenter(mView);
    }

}

File: app/src/main/java/org/xdty/callerinfo/di/modules/MainBottomModule.kt
package org.xdty.callerinfo.di.modules

import dagger.Module
import dagger.Provides
import org.xdty.callerinfo.contract.MainBottomContact.Presenter
import org.xdty.callerinfo.contract.MainBottomContact.View
import org.xdty.callerinfo.presenter.MainBottomPresenter

@Module
class MainBottomModule(private val mView: View) {
    @Provides
    internal fun provideView(): View {
        return mView
    }

    @Provides
    internal fun providePresenter(): Presenter {
        return MainBottomPresenter(mView)
    }

}